proxy/src/dns.rs

```diff
@@ -124,12 +121,15 @@ impl Resolver {
 
     // `ResolverFuture` can only be used for one lookup, so we have to clone all
     // the state during each resolution.
-    fn lookup_ip(self, &Name(ref name): &Name) -> LookupIpFuture {
+    fn lookup_ip(self, &Name(ref name): &Name)
+        -> Box<Future<Item = LookupIp, Error = ResolveError>>
```

Comment:
zomg we can use impl trait!1! (maybe that doesn't actually help, though?)

Reply:
Ah, right, that was it: this method can return `impl Future<...>` but the return value is only ever used as a field in `IpAddrListFuture::DNS`, where `impl Trait` is not allowed. So it still has to be boxed. It might be a marginally better API to return `impl Future<...>` here and box it later, though.